projects/firmware_102/src/main.c

```diff
@@ -0,0 +1,44 @@
+#include <stdbool.h>
+#include <stdint.h>  // for integer types
+#include <stdlib.h>  // for random numbers
+#include "interrupt.h"
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *count = context;
+  count->counter_a += 1;
+  LOG_DEBUG("Counter A:%d\n", count->counter_a);
+
+  if (count->counter_a % 2 == 0 && count->counter_a != 0) {
+    count->counter_b += 1;
+    LOG_DEBUG("Counter B:%d\n", count->counter_b);
+  }
+
+  soft_timer_start_millis(500,                 // timer duration
```

Comment:
These comments are not really necessary as well, but this is fine if you are just doing this for your learning. As I said above, make sure when you are actually writing code that will be a part of this repo that redundant comments like these are removed. We do this to make sure things don't get cluttered. Also, the more documentation you write, the more it needs to be managed when changes are made and the more work you need to do.

Reply:
Almost forgot to add, make sure to #define the time in milliseconds at the top. Right now it is hardcoded which is not good practice.